docs/enhancements/dynamic-fetching-of-openshift-versions.md

```diff
@@ -0,0 +1,280 @@
+        "ppc64le",
+        "s390x"
+    ],
+    "url": "quay.io/openshift-release-dev/ocp-release:4.12.0-rc.4-multi",
+    "version": "4.12.0-rc.4"
+  }
+]'
+
+DYNAMIC_OPENSHIFT_RELEASES_CONFIG='[
+  {
+    "openshift_version": "4.11",
+    "upgrade_channels": [
+        {
+          "cpu_architecture": "x86_64",
+          "channels": ["stable"]
+        }
+    ]
+  },
+  {
+    "openshift_version": "4.11",
+    "upgrade_channels": [
+      {
+        "cpu_architecture": "x86_64",
+        "channels": ["candidate", "stable"]
+      },
+      {
+        "cpu_architecture": "arm64",
+        "channels": ["candidate", "stable"]
+      }
+    ]
+  }
+]'
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Mostly conveyed in ``Risks and Mitigations``.
+
+### Risks and Mitigations
+
+We have recently seen occasions where the disk space is being exhausted due to
+the assisted-service caching a lot of different openshift installer binaries.
+It is important to make sure we don't make this problem more severe, as this
+enhancement introduces much more versions which may (or may not) cause use of
+them.
+A possible solution here is implementing the cache as a LRU-like cache, making
+sure we evict the old entries when we reach a certain capacity.
+
+We should make sure we don't consume too much memory when fetching releases.
+It's especially critical when the user queries for the different versions, but
+might happen in other cases too.
+
+We need to think of the retention policy of versions. We don't expect to have
+versions removed from the upgrade-channels API, but just in case we should only
+add versions and not mirror exactly the given result of those API responses.
+As opposed to that, when a minor release gets EOL (end-of-life) and we remove it
+from the configuration, the service should remove all the stale versions from
+the database.
+
+## Design Details [optional]
+
+If an enhancement is complex enough, design details should be included. When not
+included, reviewers reserve the right to ask for this section to be filled in to
+enable more thoughtful discussion about the enhancement and it's impact.
+
+### Open Questions
+
+* Is the database usage a real necessity? The idea is having the periodic thread
+  updating the database, and then the HTTP server is able to filter versions from
+  the database at its leisure. If there's a good way to do it in-memory and it
+  doesn't consume too much of it, it might be better.
+* When implementing the binaries LRU cache, should we prioritize the latest
+  versions of each y-release? We expect those to be the most common ones, but
+  maybe it's better to let the LRU mechanism determine what's the common versions
+  by itself?
+* How should we limit database entries? In case of a bug in the code that writes
+  way too much version entries, can we make sure to mitigate the issue on our
+  side? (without interrupting app-sre for their help on manual database
+  intervention)
+
+### UI Impact
```

Comment:
I think that the default response for the openshift-versions API should include the default(latest) version per OCP release and we should add a filter to this API with option to send ALL versions otherwise it will take lots of time to load list of 180 versions + 
@nirfarkas @jkilzi WDYT?

Reply:
Done!